piff/polynomial_interp.py

```diff
@@ -0,0 +1,361 @@
+         """
+        indices = []
+        for p in xrange(order+1):
+            for i in xrange(p+1):
+                j = p-i
+                indices.append((i,j))
+        return indices
+
+    def pack_coefficients(self, parameter_index, C):
+        """Pack the 2D matrix of coefficients used as the model fit parameters
+        into a vector of coefficients, either so this can be passed as a starting
+        point into the curve_fit routine or for serialization to file.
+
+        For subclasses, the 2D matrix format could be whatever you wanted as long
+        as initialGuess, interpolationModel, and the pack and unpack functions are 
+        consistent. The intialGuess method can return and the interpolationModel can
+        accept parameters in whatever form you like (e.g. could be a dict if you want)
+        as long as pack_coefficients can convert this into a 1D array and unpack_coefficients
+        convert it the other way.
+
+        :param parameter_index: The integer index of the parameter; the lets us 
+                                find the order of the parameter from self.
+        :param C:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+        :returns coeffs:    A 1D numpy array of coefficients of length self.nvariable
+        """
+        coeffs = numpy.zeros(self.nvariables[parameter_index])
+        for k,(i,j) in enumerate(self.indices[parameter_index]):
+            coeffs[k] = C[i,j]
+        return coeffs
+
+
+    def unpack_coefficients(self, parameter_index, coeffs):
+        """Unpack a sequence of parameters into the 2D matrix for the 
+        given parameter_index (which determines the order of the matrix)
+
+        This function is the inverse of pack_coefficients
+                           
+        :param parameter_index: The integer index of the parameter being used
+        :param coeffs:     A 1D numpy array of coefficients  of length self.nvariable
+        :returns:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+
+        """
+        k=0
+        n=self.orders[parameter_index]+1
+        C = numpy.zeros((n, n))
+        for k,(i,j) in enumerate(self.indices[parameter_index]):
+                C[i,j] = coeffs[k]
+                k+=1
+        return C
+
+    
+    def interpolationModel(self, pos, C):
+        """Generate the polynomial variation of some quantity at x and y
+        coordinates for a given coefficient matrix.
+
+        TODO: At the moment this function is expecting a numpy array of
+        shape (2,nstar) for the positions. We might want to use a galsim
+        position object instead since I think some code elsewhere in Piff
+        is expecting this.
```

Comment:
We should probably make our own Position class to encapsulate the "position" used for interpolation.  e.g. it may include the color or other such information.  And it should know which values constitute a regular (u,v) part of the position (and so need to use the max i+j order) vs. using having its own separate polynomial order.